### DISCUSSION TITLE
"Eye" trait randomly gets excluded sometimes

### DISCUSSION THREAD

**Discussion by KingNFT** [nftchef/art-engine#54] [Help + Q&A] [answered]
I'm having a problem where sometimes when I generate 50 editions the "Eye" trait gets left out, I have no NONE layer in the folder but do have some incompatible layers. See code snippet below for configuration:

// Layer order
    layersOrder: [
      { name: "Background" },
      { name: "Glyph" },
      { name: "Symbol" },
      { name: "Base" },
      { name: "Eye" },
      { name: "Tattoo" },
      { name: "Clothing" },
      { name: "Hand" },
      { name: "Necklace" },
      { name: "Hair" },
      { name: "Head" },
      { name: "Item" },
    ],

const incompatible = {
  Titan: [
    "Blue Eyes",
    "Purple Eyes",
    "Stary Eyes",
    "Yellow Eyes",
    "White Eyes",
  ],
  Aqua: ["Titan Base", "Titan Stary", "Titan Yellow"],
  Mystical: ["Titan Base", "Titan Stary", "Titan Yellow"],
  Ocean: ["Titan Base", "Titan Stary", "Titan Yellow"],
  Pale: ["Titan Base", "Titan Stary", "Titan Yellow"],
  Purple: ["Titan Base", "Titan Stary", "Titan Yellow"],
  Tinted: ["Titan Base", "Titan Stary", "Titan Yellow"],
};

![image](https://user-images.githubusercontent.com/96945030/155202007-f8384786-5d8c-4454-b991-18002f37a097.png)
![image](https://user-images.githubusercontent.com/96945030/155202324-34126fc7-29c4-4dec-a695-44a5ade450db.png)

I think having 5 eye color variations should be plentyfull for 50 editions right? I still have up to 5 editions without eyes. 
My question was how to solve it? I think the incompatible configuration is causing the problem. So maybe put the different eyes in a separate folder and make base x incompatible with folder y? I saw in the code that it is possible to exclude a directory? How do I put that in the config?

Hope this gives enough info for an answer :), if not please tell me so.

**Comment by nftchef** [comment, marked as answer]
the configuration seems OK on first look. 
Do you have the same problem when generating a smaller set, like 10?

and are you seeing the DNA Exists error ever in the console output?

Folder incompatibilities are only for weighted folders. The way to 'exclude' a full folder is to setup the branched folder nesting to go down a "different path"

**Reply by KingNFT** [reply]
This may sound stupid but I found the problem 😅. Turns out I accidently named a background layer the same as a base layer "Titan". 

Because I configured incompatibilities with the "Titan" base for the eyes it ignored normal eye layers, then went on picking a base and also ignored titan eye layers. So no eye layer was left to pick from. I renamed the background to something different and it worked! 

Interesting addition for the branched folder will check it out, thanks!

Thanks for responding so quickly! Love the algo and its extra features 👌

**Reply by nftchef** [reply]
ah, good catch! glad you figured it out!
